src/__swaps__/utils/swaps.ts

```diff
@@ -260,42 +260,55 @@ export function valueBasedDecimalFormatter({
 }): string {
   'worklet';
 
-  function calculateDecimalPlaces(): number {
-    const fallbackDecimalPlaces = MAXIMUM_SIGNIFICANT_DECIMALS;
+  function calculateDecimalPlaces(): {
+    minimumDecimalPlaces: number;
+    maximumDecimalPlaces: number;
+  } {
     if (nativePrice === 0) {
-      return fallbackDecimalPlaces;
+      return {
+        minimumDecimalPlaces: 0,
+        maximumDecimalPlaces: MAXIMUM_SIGNIFICANT_DECIMALS,
+      };
     }
+
     const unitsForOneCent = 0.01 / nativePrice;
     if (unitsForOneCent >= 1) {
-      return isStablecoin ? STABLECOIN_MINIMUM_SIGNIFICANT_DECIMALS : 0;
```

Comment:
I removed the `isStablecoin` check here because it is unnecessary; by definition, an asset with `unitsForOneCent >= 1` should not be a stablecoin, since this would imply a `nativePrice` of `0.01` or less for a stablecoin.